### ISSUE TITLE
pydantic: Invalid constructor argument not being raised by ty

### ISSUE THREAD

**Issue by AbdealiLoKo** [astral-sh/ty#4420]
### Summary

I was using a FastAPI + Sqlalchemy + Pydantic stack and was trying out `ty`
And foud that when I gave a wrong argument to pydantic, ty was not giving me a type check error

Minimal reproduction:

```python
import typing as t
from pydantic import BaseModel, ConfigDict

class WebModel(BaseModel):
    model_config = ConfigDict(from_attributes=True)

class Status(WebModel):
    status: str

_T = t.TypeVar("_T", bound=WebModel)
class Response(WebModel, t.Generic[_T]):
    result: _T

def parameterized() -> Response[Status]:
    return Response[Status](result="wrong")
```

And when I run: `uvx --with pydantic==2.13.5 ty==0.0.75 check /tmp/mre.py`

* Actual Output: **All checks passed!**
* Expected Output: `ty` should report the incorrect argument "wrong" as a `Status` was expected.

### Runtime
pydantic does give me a runtime error for it though:
```py
% .venv/bin/python /tmp/mre.py
Traceback (most recent call last):
  File "/tmp/mre.py", line 17, in <module>
    print(parameterized())
          ~~~~~~~~~~~~~^^
  File "/tmp/mre.py", line 15, in parameterized
    return Response[Status](result="wrong")
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/Users/ali/repos/ggx/.venv/lib/python3.14/site-packages/pydantic/main.py", line 263, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for Response[Status]
result
  Input should be a valid dictionary or object to extract fields from [type=model_attributes_type, input_value='wrong', input_type=str]
    For further information visit https://errors.pydantic.dev/2.13/v/model_attributes_type
```
### Debugging
When I removed the `from_attributes=True` I get the error:
```py
% uvx --with pydantic==2.13.5 ty==0.0.75 check /tmp/mre.py
error[invalid-argument-type]: Argument is incorrect
  --> /tmp/mre.py:15:29
   |
15 |     return Response[Status](result="wrong")
   |                             ^^^^^^^^^^^^^^ Expected `Status | Mapping[str, Any]`, found `Literal["wrong"]`

Found 1 diagnostic
```

### Versions
ty 0.0.75 (a553dcc1a 2026-08-26)
Python 3.14.5
Pydantic 2.13.4 and 2.13.5 tested

### Version

ty 0.0.75 (a553dcc1a 2026-08-26)

**Comment by sharkdp** [comment]
This is currently expected behavior. With `from_attributes=True`, Pydantic can construct nested models from unrelated objects with suitable attributes. For example, a field annotated as `Status` can accept an object like:

```py
class StatusSource:
    status = "ok"
```

Restricting the constructor argument to `Status | Mapping[str, Any]` (like we do for models without `from_attributes=True`) would reject valid inputs like`StatusSource`. ty currently uses `Any` for constructor parameters of models that use `from_attributes=True`, which avoids those false positives but also means it misses invalid inputs such as `"wrong"` in your example.

So I don't think there's anything we can do about it for `from_attributes=True` models with lax validation. I think your options are:
- Switch to Pydantic's strict checking, if you're okay with the change in runtime behavior, e.g.:
    ```py
    class Response(WebModel, t.Generic[_T]):
        model_config = ConfigDict(strict=True)
        result: _T
    ```
    or just for the `result` field: `result: _T = Field(strict=True)`. In that case, ty will emit _"Expected `Status`, found `Literal["wrong"]`"_.
- Remove `from_attributes=True`. In that case, ty will also report the error.
- Accept that ty can't statically verify constructor arguments on lax/non-strict mode models with `from_attributes=True`.
